lib/logger.js

```diff
@@ -310,24 +308,24 @@ configs.LOG_LEVELS.forEach(function(level) {
 });
 
 const flushAll = function(cb) {
-    var expectedCallbacks = loggers.length;
-    function callback() {
-        if (expectedCallbacks-- <= 1) cb();
-    }
-    if (!expectedCallbacks) {
-        cb();
-    } else {
-        for (var i = 0; i < loggers.length; i++) {
-            loggers[i]._flush(callback);
-        }
+    const errors = [];
+    loggers.forEach((logger) => {
+        logger._flush((err) => {
+            if (err) {
+                errors.push(err);
+            }
+        });
+    });
+    if (errors.length > 0) {
+        return cb(`The following errors happened while flushing all loggers: ${errors}`);
     }
+    return cb();
```

Comment:
I think you need to ensure `cb` is callable, since there's potential that no callback was supplied and this would result in an error.